docs/compilation.md

```diff
@@ -0,0 +1,6 @@
+TODO:
```

Comment:
I think it would be nice to add a README.md this directory with a table of contents, simply linking to the other files. Mainly so we can put a link to that file from the real README in the repo root, with the title "Detailed docs" or something.